### ISSUE TITLE
Validation error on import

### ISSUE THREAD

**Issue by benliger1987** [contentful/contentful-import#97]
I'm trying to import an exported space and am getting the following error:

>  × Validating content-file
>    → assets:3, webhooks:4 - Get further details in the error log file
>    Initialize clients
>    Checking if destination space already has any content and retrieving it
>    Apply transformations to source data
>    Push content to destination space
> 
> 
> The following 1 errors and 0 warnings occurred:
> 
> 15:30:68 - ValidationError: assets:3, webhooks:4 - Get further details in the error log file

I'm not entirely sure what's causing this. I'm fairly sure I had a successful export, I had a few warnings but as they were just warnings i'm assuming it's exported as it should. Here was the log for that:

>  √ Fetching data from space
>    √ Connecting to space (1s)
>    √ Fetching content types data (1s)
>    √ Fetching editor interfaces data (5s)
>    √ Fetching content entries data (3s)
>    √ Fetching assets data (1s)
>    √ Fetching locales data (1s)
>    √ Fetching webhooks data (1s)
>    √ Fetching roles data (1s)
>  ↓ Download assets [skipped]
>  √ Write export log file
>    √ Lookup directory to store the logs
>    √ Create log directory
>    √ Writing data to file
> ┌─────────────────────────┐
> │ Exported entities       │
> ├───────────────────┬─────┤
> │ Content Types     │ 43  │
> ├───────────────────┼─────┤
> │ Editor Interfaces │ 43  │
> ├───────────────────┼─────┤
> │ Entries           │ 531 │
> ├───────────────────┼─────┤
> │ Assets            │ 428 │
> ├───────────────────┼─────┤
> │ Locales           │ 7   │
> ├───────────────────┼─────┤
> │ Webhooks          │ 4   │
> ├───────────────────┼─────┤
> │ Roles             │ 3   │
> └───────────────────┴─────┘
> The export took a few seconds (14s)
> 
> Stored space data to json file at: C:\mydirectory\myexportedcontent.json
> 
> 
> The following 0 errors and 5 warnings occurred:
> 
> 15:26:32 - Rate limit error occurred. Waiting for 1562 ms before retrying...
> 15:26:33 - Rate limit error occurred. Waiting for 1572 ms before retrying...
> 15:26:33 - Rate limit error occurred. Waiting for 1694 ms before retrying...
> 15:26:39 - Rate limit error occurred. Waiting for 1557 ms before retrying...
> 15:26:39 - Rate limit error occurred. Waiting for 1623 ms before retrying...

**Comment by axe312ger** [comment]
Check the error log from the import tool. It should give you detailed information which exported entities might have wrong data (structure)

Fix these issues by hand and you are good :)

**Comment by benliger1987** [comment]
I have a few errors which are both titled:

> \"file\" is required

and

> \"httpBasicUsername\" must be a string

Could you recommend what this would suggest and where I can start looking?

**Comment by axe312ger** [comment]
Hey there,

file is required might be related to #96 which we will release asap

the httpBasicUsername actually should be removed via https://github.com/contentful/contentful-batch-libs/blob/master/lib/transform/transformers.js#L18-L24

You can work around the later one by removing that field manually from the source import json file. But is actually sounds like a bug since the code mentioned above should take care of this 🤔 

Best,
Benedikt

**Comment by axe312ger** [comment]
#96 is fixed in the just released v6.2.3 :)

**Comment by benliger1987** [comment]
Yes so can confirm #96 is fixed. Also I removed the httpBasicUsername field manually from my export file which resolved that error. Worth noting I did not have a completely successful import. Some of the content has synced successfully and some hasn't.

I had 160 errors and 568 warnings. Most the errors were Rate limit others were:
BadRequest: Status 400:
VersionMismatch: Status 409 - Conflict
InvalidEntry: 442 - Unprocessable Entity
UnresolvedLinks:: 442 - Unprocessable Entity

**Comment by axe312ger** [comment]
Please try to import to a fresh space to see if it still fails.

If this is the case, please open a support ticket at https://www.contentful.com/support/, give it a small writeup about the issue and mention this ticket. You may upload your import.json and error-log.json, too.

Best,
Benedikt

**Comment by dustinaleksiuk** [comment]
I'm getting very similar errors both when trying to apply a content model to an existing space using the API, as well as when exporting a space and importing it into a clean space.

**Comment by axe312ger** [comment]
@dustinaleksiuk I need more details to see the cause of the issue since these errors are all related to content and its structure. Could you maybe file a support ticket, too?

**Comment by ChaosSteffen** [comment]
This is still broken: (running 6.2.3)

```
[
  {
    "ts": "2018-04-16T12:25:20.607Z",
    "level": "error",
    "error": {
      "isJoi": true,
      "name": "ValidationError",
      "details": [
        {
          "message": "\"httpBasicUsername\" must be a string",
          "path": [
            "webhooks",
            1,
            "httpBasicUsername"
          ],
          "type": "string.base",
          "context": {
            "value": null,
            "key": "httpBasicUsername",
            "label": "httpBasicUsername"
          },
          "entity": "***"
        }
      ],
      "stacktrace": [
        "ValidationError: webhooks:1 - Get further details in the error log file",
        "Object.exports.process (/usr/local/lib/node_modules/contentful-import/node_modules/joi/lib/errors.js:190:19)",
        "internals.Object._validateWithOptions (/usr/local/lib/node_modules/contentful-import/node_modules/joi/lib/types/any/index.js:669:31)",
        "module.exports.internals.Any.root.validate (/usr/local/lib/node_modules/contentful-import/node_modules/joi/lib/index.js:139:23)",
        "assertPayload (/usr/local/lib/node_modules/contentful-import/dist/utils/validations.js:43:30)",
        "Task.task (/usr/local/lib/node_modules/contentful-import/dist/index.js:101:38)",
        "Promise.resolve.then.then.skipped (/usr/local/lib/node_modules/contentful-import/node_modules/listr/lib/task.js:168:30)",
        "<anonymous>",
        "process._tickCallback (internal/process/next_tick.js:182:7)",
        "Function.Module.runMain (internal/modules/cjs/loader.js:697:11)",
        "startup (internal/bootstrap/node.js:201:19)",
        "bootstrapNodeJSCore (internal/bootstrap/node.js:516:3)"
      ]
    }
  }
]
```

**Comment by Khaledgarbaya** [comment]
hey @ChaosSteffen,
can you check the webhook structure in your exported data.
I am more interested in the `httpBasicUsername` does it exist? if it exist does it have the value null?

**Comment by ChaosSteffen** [comment]
Hi @Khaledgarbaya, it is null.

```
"httpBasicUsername": null,
```

**Comment by Khaledgarbaya** [comment]
@ChaosSteffen,
Ok that's a bug in the import tool validation it does not allow null I will release a fix today.

Best,
Khaled

**Comment by ChaosSteffen** [comment]
Awesome, thank you!

**Comment by Khaledgarbaya** [comment]
This is fixed in [v7.0.1](https://github.com/contentful/contentful-import/releases/tag/v7.0.1)
